motoman_driver/MotoPlus/SimpleMessage.c

```diff
@@ -151,6 +151,12 @@ int Ros_SimpleMsg_MotionReply(SimpleMsg* receiveMsg, int result, int subcode, Si
 		replyMsg->body.motionReply.sequence = receiveMsg->body.jointTrajDataEx.sequence;
 		replyMsg->body.motionReply.command = ROS_MSG_MOTO_JOINT_TRAJ_PT_FULL_EX;
 	}
+	if (receiveMsg->header.msgType == ROS_MSG_MOTO_SELECT_TOOL)
+	{
+		replyMsg->body.motionReply.groupNo = ctrlGrp;
+		replyMsg->body.motionReply.sequence = -1;

```

Comment:
Is `-1` a proper value for `sequence` here in a reply? The other messages get the incoming `sequence` copied into the reply.

Reply:
The other messages are replying to a specific trajectory command; each of which has a sequence identifier in the trajectory. The SELECT_TOOL command doesn't have a sequence identifier, so -1 is used.

Reply:
I think it may still make sense to copy the incoming sequence nr, as it will allow clients to identify which response belongs to which request.

There is very little support for asynchronous interaction with the driver, but this is small thing to not make it even more difficult than it already is.

Reply:
You're right. I was thinking it was a trajectory sequencer, rather than a packet sequencer. But that's a different command...